rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -3546,4 +3550,69 @@ public ResponseEntity<?> createDuplicateProjectWithDependencyNetwork(
+                                required = true,
+                                example = "12345",
+                                schema = @Schema(type = "string")
+                            )
+                        },
+            responses = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "License information successfully added to linked releases.",
+                            content = @Content(
+                                mediaType = "application/hal+json",
+                                schema = @Schema(type = "object", implementation = JSONObject.class),
+                                examples = @ExampleObject(
+                                        value = "{\n  \"releasesWithSingleCLI\": [\"Release1\", \"Release2\"],\n  \"releasesWithMultipleCLI\": [\"Release3\"],\n  \"STATUS\": \"SUCCESS\"\n}"
+                                )
+                            )
+                        ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Error occurred while processing license information for linked releases.",
+                            content = @Content(
+                                mediaType = "application/json",
+                                examples = @ExampleObject(
+                                    value = "{\n  \"error\": \"Error adding license info to linked releases.\"\n}"
+                                )
+                            )
+                        )
+          }
+    )
+    @PostMapping(value = PROJECTS_URL + "/{id}/addLinkedRelesesLicenses")
+    public ResponseEntity<String> addLicenseToLinkedReleases(
+            @Parameter(description = "Project ID", example = "376576")
+            @PathVariable("id") String projectId) {
+        try {
+            User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+            Project project = projectService.getProjectForUserById(projectId, sw360User);
+
+            if (project == null) {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Project not found.");
+            }
+
+            RequestStatus requestStatus = projectService.addLicenseToLinkedReleases(projectId, sw360User);
+
+            if (requestStatus == RequestStatus.SUCCESS) {
+                return ResponseEntity.ok("License information successfully added to linked releases.");
+            }
+
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body("Error adding license info to linked releases.");
+
+        } catch (TException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Thrift error: " + e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Unexpected error: " + e.getMessage());
```

Comment:
The exception should not be part of response. Please do following instead.

`throw new ResourceClassNotFoundException("Internal server error.", e);`

This will return a 500 response and put the exception information in the application log instead of the response body.